### PR TITLE
Added sDefendPoint

### PR DESCRIPTION
**Pull request by ashishkg0022**
it assumes a class named obstacle to be defined with members x, y and radius

### CODE REVIEW

File: scripts/skills/sDefendPoint.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import math
+
+def execute(param, state, bot_id):
+    DEFEND_RADIUS = 50.0
+    ob=Vector2D()
+    botPos=Vector2D(state.homePos[bot_id].x, state.homePos[bot_id].y)
+    point_ball_angle = atan2(state.ballPos.y-param.DefendPointP.y,state.ballPos.x-param.DefendPointP.x)
+    dpoint =Vector2D(param.DefendPointP.x + (param.DefendPointP.radius)*cos(point_ball_angle) , param.DefendPointP.y + (param.DefendPointP.radius)*sin(point_ball_angle))
+    obs = []
+    o=obstacle() #assuming a class that defines obstacle
```
Comment (annimesh2809) [inline scripts/skills/sDefendPoint.py]: The [Obstacle class](https://github.com/krssg-ssl/navigation_py/blob/master/scripts/navigation/obstacle.py#L3) definiton is present in [krssg-ssl/navigation_py/scripts/navigation/obstacle.py](https://github.com/krssg-ssl/navigation_py/blob/master/scripts/navigation/obstacle.py). You can ignore the unused parameters (x2, y2).

File: scripts/skills/sDefendPoint.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import skill_node
+import math
+
+def execute(param, state, bot_id):
+    DEFEND_RADIUS = 50.0
+    ob=Vector2D()
```
Comment (annimesh2809) [inline scripts/skills/sDefendPoint.py]: You need to import [geometry.py](https://github.com/krssg-ssl/plays_py/blob/master/scripts/utils/geometry.py) to use `Vector2D`!

File: scripts/skills/sDefendPoint.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    botPos=Vector2D(state.homePos[bot_id].x, state.homePos[bot_id].y)
+    point_ball_angle = atan2(state.ballPos.y-param.DefendPointP.y,state.ballPos.x-param.DefendPointP.x)
+    dpoint =Vector2D(param.DefendPointP.x + (param.DefendPointP.radius)*cos(point_ball_angle) , param.DefendPointP.y + (param.DefendPointP.radius)*sin(point_ball_angle))
+    obs = []
+    o=obstacle() #assuming a class that defines obstacle  
+    for i in  range (0 , len(state.homeDetected)) :
```
Comment (annimesh2809) [inline scripts/skills/sDefendPoint.py]: I believe that `len(state.homeDetected)` will always return 6 (see [this](https://github.com/krssg-ssl/belief_state/blob/master/src/node.cpp#L337)). What you are looking for is something like this:
```python
for i,bot in enumerate(state.homePos):
    if state.homeDetected[i]:
        obs.append(Obstacle(bot.x, bot.y, 0, 0, 3*BOT_RADIUS))
```

File: scripts/skills/sDefendPoint.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        o.x = state.homePos[i].x
+        o.y = state.homePos[i].y
+        o.radius = 3 * BOT_RADIUS
+        obs.append(o)
+
+
+    for i in range (len(state.homeDetected) , len(state.homeDetected) + len(state.awayDetected) ):
```
Comment (annimesh2809) [inline scripts/skills/sDefendPoint.py]: Same as above

**Review by annimesh2809 — CHANGES_REQUESTED** [review]

File: scripts/skills/sDefendPoint.py (hunk 2 above)
Comment (ashishkg0022) [reply scripts/skills/sDefendPoint.py]: geometry.py is used by almost every skill ! can't we import geometry.py in skill_node only !

Comment (annimesh2809) [reply scripts/skills/sDefendPoint.py]: Yes, that would be better.

**Pull request merged**